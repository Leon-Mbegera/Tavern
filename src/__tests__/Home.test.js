import React from 'react';
import { BrowserRouter, MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { cleanup, render, screen } from '@testing-library/react';
import store from '../store';
import SearchBar from '../components/Search';

afterEach(cleanup);

describe('Home component', () => {
  test('should render the search component', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Provider store={store}>
          <SearchBar />
        </Provider>
      </BrowserRouter>,
    );
    expect(
      asFragment(
        <Provider store={store}>
          <SearchBar />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });
});
