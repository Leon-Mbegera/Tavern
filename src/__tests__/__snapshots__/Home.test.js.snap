// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render the search component 1`] = `
<DocumentFragment>
  <div
    class="sub-header container-fluid d-flex flex-direction-column justify-content-between"
  >
    <div
      class="d-flex flex-direction-column offer"
    >
      <h4
        class="fw-bold"
      >
        NOVEMBER BLACK FRIDAY SALES
      </h4>
      <p
        class="discount py-1"
      >
        Up to 50% off
      </p>
      <p
        class="live-now blink"
      >
        LIVE NOW
      </p>
    </div>
    <div
      class="form-outline my-3 Searchbar"
    >
      <input
        class="form-control"
        placeholder="Search cocktail by name.."
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
